File: src/main/java/com/example/venka/javaspringsparkdemo/config/MongoConfiguration.java
package com.example.venka.javaspringsparkdemo.config;

import com.mongodb.MongoClient;
import com.mongodb.client.MongoDatabase;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.autoconfigure.mongo.MongoProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class MongoConfiguration {

    @Bean
    public MongoDatabase getMongoDatabase(@Autowired MongoProperties mongoProperties) {
        return new MongoClient(mongoProperties.getHost(), mongoProperties.getPort())
                .getDatabase(mongoProperties.getDatabase());
    }
}

File: src/main/kotlin/com/example/venka/kotlinspringsparkdemo/config/MongoConfiguration.kt
package com.example.venka.kotlinspringsparkdemo.config

import com.mongodb.MongoClient
import com.mongodb.client.MongoDatabase
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.autoconfigure.mongo.MongoProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class MongoConfiguration {

    @Bean
    fun getMongoDatabase(@Autowired mongoProperties: MongoProperties): MongoDatabase {
        return MongoClient(mongoProperties.host, mongoProperties.port!!)
                .getDatabase(mongoProperties.database)
    }
}
